### DISCUSSION TITLE
Heatmap

### DISCUSSION THREAD

**Discussion by pbpayal** [satijalab/seurat#4287] [Q&A] [answered]
Hello,

I created a heatmap using the DoHeatmap function for 1 cluster out of the 5 clusters of my dataset. This is an integrated dataset with treatment and control groups. So the DefaultAssay is set to "integrated". I have selected some interesting genes for my dataset from the DEGs that I found using FindMarkers. I wanted to plot just those genes in the heatmap. So first I subsetted the clutser I wanted and then selected the genes I wanted to showcase and its giving a nice heatmap. 

I am having trouble interpreting the heatmap as I am used to seeing bulk RNA heatmaps. These heatmaps are a bit different.  Its written that the default data slot is "scale.data". So I tried to extract the scaled data and tried to do average of the expression values but then the foldchange or expression that's showing up is very different from the heatmap. **So I am wondering how is the data transformed from 'scale.data' in the DoHeatmap function to project the heatmaps?**

`cluster_1_subset <- subset(immune.combined.clusters, idents = c("1_control", "1_treatment"), invert = FALSE)`

`write.csv(cluster_1_subset@assays$integrated@scale.data,"cluster_1_subset_heatmap_data.csv")`

`cluster1_features = c("ENSMUSG00000003974","ENSMUSG00000004366","ENSMUSG00000006494","ENSMUSG00000006782","ENSMUSG00000008999","ENSMUSG00000015312","ENSMUSG00000015461","ENSMUSG00000021466","ENSMUSG00000021760","ENSMUSG00000023927","ENSMUSG00000024182","ENSMUSG00000026787","ENSMUSG00000027004","ENSMUSG00000029231","ENSMUSG00000030108","ENSMUSG00000030287","ENSMUSG00000031343","ENSMUSG00000031425","ENSMUSG00000041235","ENSMUSG00000043969","ENSMUSG00000044167","ENSMUSG00000074218")`

`DoHeatmap(cluster_1_subset, features = cluster1_features, size = 3, angle = 0,  draw.lines = T)`

Attached is the heatmap I am getting.

![Cluster1_heatmap](https://user-images.githubusercontent.com/20340380/112338489-3741b180-8c95-11eb-8104-1eae09bec30c.png)

Disclaimer - This dataset has very low number of cells in the range of 50 -100 cells per cluster as the data was produced using FluidigmC1 4-5 years back.

Any explanation will be very helpful. 

Thanks,
Payal

**Comment by timoast** [comment, marked as answer]
1. We don't recommend using the integrated data for visualization of differential expression
2. The transformation for scaled data is scaling (divide by standard deviation) and centering (subtract mean) the normalized data. This equates to cells having expression values equal to the average value across the dataset having a values of zero, and deviations above or below the mean expression will be positive or negative.

>  am having trouble interpreting the heatmap as I am used to seeing bulk RNA heatmaps.

Each column is a cell and each row is a gene. Cells are grouped by a grouping variable (`1_control` and `1_treatment` in your case)
